lib/ask/runtime/step.ex

```diff
@@ -165,18 +165,12 @@ defmodule Ask.Runtime.Step do
       "numeric" ->
         # Only send numDigits if the min and max values have the same string length,
         # also taking into account the values of refusal responses
-        refusal = step["refusal"]
-
-        values = if refusal do
-          refusal["responses"]["ivr"]
-        else
-          []
-        end
-
         min_value = step["min_value"]
         max_value = step["max_value"]
+        refusal_values = fetch(:refusal, step, "ivr", language) || []
+
         if min_value && max_value do
-          values = [min_value, max_value | values]
+          values = [min_value, max_value | refusal_values]
```

Comment:
Just one more suggestion (because it is misleading at first sight)
If I got it right, the `Enum.map` (in line 174) is counting the digits, so maybe we could re-write it as:
```
values = [min_value, max_value | refusal_values]
|> Enum.map(& (&1 |> Integer.digits |> Enum.count))
|> Enum.uniq
```

An example of the `Integer.digits |> Enum.count` composition
```console
iex(1)> [1, 2, 42] |> Enum.map(& (&1 |> Integer.digits |> Enum.count))
[1, 1, 2]
```

I hope that with this change it makes the code more readable.

Reply:
It's usually a number, but it doesn't have to, for example it can be `#` or `*`.